File: GetData.R
install.packages("mongolite")
library(mongolite)
connection <- mongo(collection = "truck_1",
                    db = "TruckData",
                    url = "mongodb://user:password@mongo:27017")

data <- connection$find()
dataAggr <- connection$aggregate('[
    {
        "$match": {
            "timestamp": {
                "$gte": "2021-05-23"
            }
        }
    }
]')

File: GetData.py
from pymongo import MongoClient
import pandas as pd

client = MongoClient("mongodb://user:password@mongo:27017")
db = client["TruckData"]
col = db["truck_1"]

data = pd.DataFrame(col.find())
pipeline = [
    {
        '$match': {
            'timestamp': {
                '$gte': '2022-01-07'
            }
        }
    }
]

df = pd.DataFrame(col.aggregate(pipeline))
df